airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/http_requester.py

```diff
@@ -553,3 +486,206 @@ def _try_get_error(value: Any) -> Any:
+        if self.request_options_provider is None:
+            self._request_options_provider = InterpolatedRequestOptionsProvider(config={}, parameters=parameters)
+        elif isinstance(self.request_options_provider, dict):
+            self._request_options_provider = InterpolatedRequestOptionsProvider(config=self.config, **self.request_options_provider)
+        else:
+            self._request_options_provider = self.request_options_provider
+
+        super().__post_init__(parameters)
+
+    # We are using an LRU cache in should_retry() method which requires all incoming arguments (including self) to be hashable.
+    # Dataclasses by default are not hashable, so we need to define __hash__(). Alternatively, we can set @dataclass(frozen=True),
+    # but this has a cascading effect where all dataclass fields must also be set to frozen.
+    def __hash__(self) -> int:
+        return hash(tuple(self.__dict__))
+
+    def get_url_base(self) -> str:
+        return os.path.join(self._url_base.eval(self.config), "")
+
+    def get_path(
+        self, *, stream_state: Optional[StreamState], stream_slice: Optional[StreamSlice], next_page_token: Optional[Mapping[str, Any]]
+    ) -> str:
+        kwargs = {"stream_state": stream_state, "stream_slice": stream_slice, "next_page_token": next_page_token}
+        path = str(self._path.eval(self.config, **kwargs))
+        return path.lstrip("/")
+
+    def get_request_params(
+        self,
+        *,
+        stream_state: Optional[StreamState] = None,
+        stream_slice: Optional[StreamSlice] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> MutableMapping[str, Any]:
+        return self._request_options_provider.get_request_params(
+            stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token
+        )
+
+    def get_request_headers(
+        self,
+        *,
+        stream_state: Optional[StreamState] = None,
+        stream_slice: Optional[StreamSlice] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> Mapping[str, Any]:
+        return self._request_options_provider.get_request_headers(
+            stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token
+        )
+
+    # fixing request options provider types has a lot of dependencies
+    def get_request_body_data(  # type: ignore
+        self,
+        *,
+        stream_state: Optional[StreamState] = None,
+        stream_slice: Optional[StreamSlice] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> Union[Mapping[str, Any], str]:
+        return (
+            self._request_options_provider.get_request_body_data(
+                stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token
+            )
+            or {}
+        )
+
+    # fixing request options provider types has a lot of dependencies
```

Comment:
can you expand on this comment? specifically, what isn't fixed and why it's difficult to fix it?

Reply:
This comment already existed too, but it seems that `type: ignore` is not needed here, so I removed this comment